src/player.cpp

```diff
@@ -1032,35 +1032,10 @@ void player::update_bodytemp()
     // Sunlight
     const int sunlight_warmth = g->is_in_sunlight( pos() ) ? 0 :
                                 ( g->weather == WEATHER_SUNNY ? 1000 : 500 );
-    // Fire at our tile
-    const int fire_warmth = bodytemp_modifier_fire();
-
-    // Cache fires to avoid scanning the map around us bp times
-    // Stored as intensity-distance pairs
-    std::vector<std::pair<int, int>> fires;
-    fires.reserve( 13 * 13 );
-    int best_fire = 0;
-    for( const tripoint &dest : g->m.points_in_radius( pos(), 6 ) ) {
-        int heat_intensity = 0;
-
-        int ffire = g->m.get_field_strength( dest, fd_fire );
-        if( ffire > 0 ) {
-            heat_intensity = ffire;
-        } else if( g->m.tr_at( dest ).loadid == tr_lava ) {
-            heat_intensity = 3;
-        }
-        if( heat_intensity == 0 || !g->m.sees( pos(), dest, -1 ) ) {
-            // No heat source here
-            continue;
-        }
-        // Ensure fire_dist >= 1 to avoid divide-by-zero errors.
-        const int fire_dist = std::max( 1, square_dist( dest, pos() ) );
-        fires.emplace_back( std::make_pair( heat_intensity, fire_dist ) );
-        if( fire_dist <= 1 ) {
-            // Extend limbs/lean over a single adjacent fire to warm up
-            best_fire = std::max( best_fire, heat_intensity );
-        }
-    }
+    // Hot air from a heat source
+    // times 50 to convert to weather.h BODYTEMP scale
+    const int fire_warmth = g->get_convection_temperature( pos() ) * 50;
+    const int best_fire = g->get_heat_radiation( pos(), true );
```

Comment:
If `game::get_heat_radiation` and `game::get_convection_temperature` are made into free functions change calls to `get_heat_radiation( g->m, pos(), true );` and `get_convection_temperature( g->m, pos() );`